app/uk/gov/hmrc/customs/emailfrontend/controllers/EmailConfirmedController.scala

```diff
@@ -64,26 +66,29 @@ class EmailConfirmedController @Inject()(identify: IdentifierAction,
 
   private def updateEmail(details: EmailDetails)(implicit request: AuthenticatedRequest[_]): Future[Result] = {
     val timestamp = dateTimeService.nowUtc()
+
+    lazy val viewModel = for {
+      _ <- save4LaterService.saveEmail(request.user.internalId, details.copy(timestamp = Some(timestamp)))
+
+      _ <- customsDataStoreService.storeEmail(
+        EnrolmentIdentifier("EORINumber", request.user.eori), details.newEmail, timestamp)
+
+      maybeReferrerName <- save4LaterService.fetchReferrer(request.user.internalId)
+      journeyType <- save4LaterService.fetchJourneyType(request.user.internalId)
+    } yield EmailVerifiedOrChangedViewModel(
+      details.newEmail,
+      maybeReferrerName.map(_.continueUrl),
+      journeyType.fold {
+        logger.warn("Unable to determine journey type!")
```

Comment:
Are we supposed to be logging this warning every time we call this? I can't see how it could be anything but this?

Reply:
What do you think?

Reply:
Just for context the current behaviour is to show the user a problem with service page, so maybe we should stick with that? (Even though their email could still be verified/updated!)

Reply:
Pending your input, I am going to change it back to the previous behaviour for now on the pretext that I create a tech debt ticket to address this more properly.